clang/lib/AST/ExprConstant.cpp

```diff
@@ -6781,7 +6781,7 @@ static bool HandleConstructorCall(const Expr *E, const LValue &This,
       // and make sure we've initialized every step along it.
       auto IndirectFieldChain = IFD->chain();
       for (auto *C : IndirectFieldChain) {
-        FD = cast<FieldDecl>(C);
+        FD = cast<FieldDecl>(C)->getCanonicalDecl();
```

Comment:
I'm slightly tired to see a lot of random "getCanonicalDecl()"  in the code... but it is not your fault. Let it go.

Reply:
Instead of this change, this should just do `!declaresSameEntity(Value->getUnionField(), FD)  instead of `Value->getUnionField() != FD` a few lines below (outside of GitHub context).

Reply:
> Instead of this change, this should just do `!declaresSameEntity(Value->getUnionField(), FD)` instead of `Value->getUnionField() != FD` a few lines below (outside of GitHub context).

Done, it is basically the same getCanonicalDecl but better for explaining the intent.